driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/creat.bpf.c

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2022 The Falco Authors.
+ *
+ * This file is dual licensed under either the MIT or GPL 2. See MIT.txt
+ * or GPL2.txt for full copies of the license.
+ */
+
+#include <helpers/interfaces/variable_size_event.h>
+
+/*=============================== ENTER EVENT ===========================*/
+
+SEC("tp_btf/sys_enter")
+int BPF_PROG(creat_e,
+	     struct pt_regs *regs,
+	     long id)
+{
+	struct auxiliary_map *auxmap = auxmap__get();
+	if(!auxmap)
+	{
+		return 0;
+	}
+
+	auxmap__preload_event_header(auxmap, PPME_SYSCALL_CREAT_E);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	/* Parameter 1: name (type: PT_FSPATH) */
+	unsigned long name_pointer = extract__syscall_argument(regs, 0);
+	auxmap__store_charbuf_param(auxmap, name_pointer, USER);
+
+	/* Parameter 2: mode (type: PT_UINT32) */
+	unsigned long mode = extract__syscall_argument(regs, 1);
```

Comment:
Silly question, why are we using `unsigned long` and not `u32`?

Reply:
This is because `open_modes_to_scap` wants an unsigned long, I agree with you that it's quite confusing but I wouldn't change it here since it could cause strange verifier issues like [here](https://github.com/falcosecurity/libs/pull/297/files). But yes we need a cleanup also in the old probe!